target_chains/cosmwasm/tools/src/pipeline.ts

```diff
@@ -0,0 +1,216 @@
+import { readFileSync, writeFileSync, existsSync, mkdirSync } from "fs";
+import { createInterface } from "readline";
+
+// This function lets you write a question to the terminal
+// And returns you the response of the user
+function readLineAsync(msg: string) {
+  const readline = createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  return new Promise((resolve) => {
+    readline.question(msg, (userRes) => {
+      resolve(userRes);
+    });
+  });
+}
+
+type StateStore<T = any> = {
+  [stepId: string]: T;
+};
+// StateManager helps in getting and setting the state locally
+// It manipulates data in-memory and once the consumer has finished manipulating it
+// They need to commit the data to permanent storage using the commit method
+class StateManager<T> {
+  private readonly fileExt = ".json";
+  private readonly dirPath = "./tmp";
+
+  private readonly store: StateStore<T>;
+  private readonly filePath: string;
+
+  constructor(
+    private readonly stageId: string,
+    private readonly version: string
+  ) {
+    // dir check
+    if (!existsSync(this.dirPath)) {
+      mkdirSync(this.dirPath);
+    }
+
+    this.filePath = `${this.dirPath}/${this.stageId}-${this.version}${this.fileExt}`;
```

Comment:
i think this path structure should be defined at a higher level in the stack (like in the CLI, not in the framework itself)